src/modules/sync/utils/htmlRenderers.ts

```diff
@@ -0,0 +1,638 @@
+    "allowed_text_blocks",
+    "allowed_formatting",
+    "allowed_table_blocks",
+    "allowed_table_text_blocks",
+    "allowed_table_formatting",
+  ];
+
+  const allowsAll = emptyAllowsAll.includes(property) && Array.isArray(value) && value.length === 0;
+
+  return `<div>${property}: ${allowsAll ? "<strong>all</strong>" : getValueOrIdentifier(value)}</div>`;
+};
+
+const renderAddedTaxonomyTerm = (
+  taxonomy: TaxonomyModels.IAddTaxonomyRequestModel,
+) => `<div class="element">${taxonomy.name}</div>`;
+
+const renderAddedTaxonomyData = createRenderAddedEntityData(renderAddedTaxonomyTerm);
+
+const renderAddedTypeOrSnippetData = createRenderAddedEntityData(renderAddedElement);
+
+const renderReplaceOpValue = (
+  value: unknown,
+): ReplacePatchOperationValue | null => {
+  switch (typeof value) {
+    case "string":
+    case "boolean":
+    case "number":
+      return value;
+    case "object":
+      if (isCountLimitation(value)) {
+        return `${value.condition}: ${value.value}`;
+      }
+      if (isObjectReference(value)) {
+        return value.codename;
+      }
+      if (isObjectReferenceArray(value)) {
+        return value.map((r) => r.codename).join(", ");
+      }
+      if (isExternalIdReference(value)) {
+        return value.external_id;
+      }
+      if (isExternalIdReferenceArray(value)) {
+        return value.map((r) => r.external_id).join(", ");
+      }
+      if (isDependsOn(value)) {
+        return `${value.element.codename} ${value.snippet?.codename ? "of snippet " + value.snippet.codename : ""}`;
+      }
+      if (isDefaultValue(value)) {
+        return renderReplaceOpValue(value.global.value);
+      }
+      if (isMaximumTextLength(value)) {
+        return `${value.value} ${value.applies_to}`;
+      }
+      if (isValidationRegex(value)) {
+        return `<p><strong>Regex:</strong> ${value.regex}<strong></p><p>Flags:</strong> ${
+          value.flags ?? "—"
+        }</p><p><strong>IsActive:</strong> ${value.is_active || "—"}</p><p><strong>Validation message:</strong> ${
+          value.validation_message ?? "—"
+        }</p>`;
+      }
+      return getValueOrIdentifier(value);
+    default:
+      return null;
+  }
+};
+
+const renderDeletedEntity = (entityCodename: string) =>
+  `<div class="entity-detail"><div class="entity-name removed">${entityCodename}</div></div>`;
+
+const renderAddedTypesOrSnippetsSectionData = <T extends RequiredCodename<TypeOrSnippet>>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "added">,
+) => diff.added.map(a => renderAddedTypeOrSnippetData(a, a.elements)).join("\n");
+
+const renderAddedTaxonomiesSectionData = <T extends TaxonomyModels.IAddTaxonomyRequestModel>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "added">,
+) => diff.added.map(a => renderAddedTaxonomyData(a, a.terms)).join("\n");
+
+const renderDeletedEntitiesSectionData = <T extends { codename: string }>(
+  diff: Pick<DiffObject<RequiredCodename<T>>, "deleted">,
+) =>
+  `${
```

Comment:
this does not have to be in string formatting :) Also It can be one lined :D 
![image](https://github.com/kontent-ai/data-ops/assets/54802833/eab86350-12b3-4901-9a62-bc194f09e237)

Reply:
fixed